### ISSUE TITLE
Draw arc, pieslice and ellipse silently fail with some coordinates

### ISSUE THREAD

**Issue by JamesGKent** [python-pillow/Pillow#3029]
if i try and use the arc, pieslice or ellipse methods of a Draw object with some sets of coordinates no error is raised but nothing is drawn either.
this occurs when one or both axis coordinates decreases between the first and second coord of the bounding box.
eg:
```python
draw.arc((0,0,10,10), 0, 90, 'red') # works
draw.arc((10,0,0,10), 0, 90, 'red') # doesn't work
draw.arc((0,10,10,0), 0, 90, 'red') # doesn't work
draw.arc((10,10,0,0), 0, 90, 'red') # doesn't work
```

I'm using python3.4 and pillow 5.0.0

I have no issue with the API requiring the coordinates in the correct order, but I couldn't find any documentation stating that it was required, and obviously no error was raised either.
I think it needs documenting and maybe raise an error too?

**Comment by radarhere** [comment]
https://pillow.readthedocs.io/en/latest/reference/ImageDraw.html?highlight=imagedraw#PIL.ImageDraw.PIL.ImageDraw.ImageDraw.arc
> Parameters: xy – Two points to define the bounding box. Sequence of [(x0, y0), (x1, y1)] or [x0, y0, x1, y1].

To check that you've seen this, and to mention it here as part of the discussion, you think this doesn't adequately explain the argument?

**Comment by JamesGKent** [comment]
it does but it could be a lot clearer, at a quick glance it could easily be interpreted as x and y of first coord and x and y of second, with no clear restrictions as to needing to in ascending order.

**Comment by radarhere** [comment]
Okay, let me try again. I found this? https://pillow.readthedocs.io/en/latest/handbook/concepts.html#coordinate-system

**Comment by hugovk** [comment]
Here's the same code, but a fuller example with bigger numbers and different colours. Only the white arc is drawn:
```python
from PIL import Image, ImageDraw

im = Image.new("RGB", (120, 120))
draw = ImageDraw.Draw(im)

draw.arc((0, 0, 100, 100), 0, 90, 'white')  # works
draw.arc((100, 0, 0, 100), 0, 90, 'red')    # doesn't work
draw.arc((0, 100, 100, 0), 0, 90, 'green')  # doesn't work
draw.arc((100, 100, 0, 0), 0, 90, 'blue')   # doesn't work

im.show()
```
![image](https://user-images.githubusercontent.com/1324225/37245079-87c98bce-249b-11e8-827a-f69db3710a07.png)

It doesn't matter if you do `[(x0, y0), (x1, y1)]` or `[x0, y0, x1, y1]`, we get the same result with:
```python
draw.arc(((0, 0), (100, 100)), 0, 90, 'white')  # works
draw.arc(((100, 0), (0, 100)), 0, 90, 'red')    # doesn't work
draw.arc(((0, 100), (100, 0)), 0, 90, 'green')  # doesn't work
draw.arc(((100, 100), (0, 0)), 0, 90, 'blue')   # doesn't work
```

Here are the four coordinates:
<img width=50% src=https://user-images.githubusercontent.com/1324225/37245103-ed9f2968-249b-11e8-8657-15287974de6c.png>

Opposite corners are used in different combinations, and I might expect different arcs, but it's the same square and I might expect some sort of arc drawn for each.

Arcs are drawn using this `ellipse` C function:
https://github.com/python-pillow/Pillow/blob/81c88b1ead5343d9778a82977c9c6c1150fd9738/src/libImaging/Draw.c#L735-L838
Which is also used for `ImageDraw.ellipse`, `chord` and `pieslice`, which similarly only draw the white one.

And right at the top it calculates the `width` and `height`, returning 0 if they're negative, therefore assuming `(x1, y1)` is greater than `(x0, y0)`.

https://github.com/python-pillow/Pillow/blob/81c88b1ead5343d9778a82977c9c6c1150fd9738/src/libImaging/Draw.c#L750-L753

Commenting out that `if` gives us this:
![image](https://user-images.githubusercontent.com/1324225/37245339-24960146-249f-11e8-9ff8-2a7b8aa5a444.png)

We could "fix" it like this, but remember each arc is drawing from 0 to 90 degrees, so it's kind of twisting depending on how the "direction" the bounding box is defined. I think this could be get quite confusing, and might also break existing code that is written for the current design, which has been like this since _at least_ PIL 1.1.1 (released in 2000).

So I propose we better document that the bounding box for these four methods is a "Sequence of `[(x0, y0), (x1, y1)]` or `[x0, y0, x1, y1]` where `x1 >= x0` and `y1 >= y0`".

**Comment by hugovk** [comment]
Re-opening this so it can be closed by PR #3080 :)
